d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -1125,13 +1156,39 @@ void onRemoval(String resourceName)
       }
     }
 
-    private void onAllResourcesProcessed()
+    @VisibleForTesting
+    void onAllResourcesProcessed()
     {
+      if (_isFirstFetch)
+      {
+        _isFirstFetch = false;
+      }
+
       for (WildcardResourceWatcher watcher : _watchers)
       {
         watcher.onAllResourcesProcessed();
       }
     }
+
+    private void resetSubscribedAt()
+    {
+      _subscribedAt = _clock.currentTimeMillis();
+    }
```

Comment:
Should this function also reset `_isFirstFetch`? Otherwise on a reconnect, the `getActiveInitialWaitTimeMillis` won't reflect it